### PR TITLE
gateio support postOnly flag

### PR DESCRIPTION
**Pull request by frosty00**
also fix error when using gt to pay for fees

### CODE REVIEW

File: js/gateio.js [hunk 1]
```diff
@@ -2045,9 +2041,15 @@ module.exports = class gateio extends Exchange {
         const stopPrice = this.safeNumber (params, 'stopPrice');
         let methodTail = 'Orders';
         const reduceOnly = this.safeValue2 (params, 'reduce_only', 'reduceOnly');
-        const defaultTimeInForce = this.safeValue2 (params, 'tif', 'time_in_force', 'gtc');
+        const defaultTimeInForce = this.safeValue2 (params, 'tif', 'time_in_force');
         let timeInForce = this.safeValue (params, 'timeInForce', defaultTimeInForce);
-        params = this.omit (params, [ 'stopPrice', 'reduce_only', 'reduceOnly', 'tif', 'time_in_force', 'timeInForce' ]);
+        const postOnly = this.safeValue (params, 'postOnly', false);
+        if (postOnly) {
+            timeInForce = 'poc';
```
Comment (kroitor) [inline js/gateio.js:2048]: According to gateio's docs, the `'poc'` value is not supported for conditional orders. Can you test that?

Comment (frosty00) [reply js/gateio.js:2048]: where does it say that?

Comment (kroitor) [reply js/gateio.js:2048]: Here (it does not list `poc`):

- https://www.gate.io/docs/developers/apiv4/en/#create-a-price-triggered-order


Screenshot 2021-11-20 at 14 07 08

- https://www.gate.io/docs/developers/apiv4/en/#create-a-price-triggered-order-2

Screenshot 2021-11-20 at 14 06 14

- https://www.gate.io/docs/developers/apiv4/en/#create-a-price-triggered-order-3

Screenshot 2021-11-20 at 14 07 48

File: js/gateio.js [hunk 2]
```diff
@@ -1865,14 +1861,14 @@ module.exports = class gateio extends Exchange {
             feeCost = this.safeNumber (trade, 'fee');
         } else {
             feeCurrency = 'GT';
-            feeCost = this.parseNumber (gtFee);
+            feeCost = gtFee;
```
Comment (kroitor) [inline js/gateio.js:1864]: https://app.travis-ci.com/github/ccxt/ccxt/builds/242134941#L5773

File: js/gateio.js [hunk 3]
```diff
@@ -1847,32 +1847,33 @@ module.exports = class gateio extends Exchange {
...
-            feeCost = this.safeNumber (trade, 'fee');
-        } else {
-            feeCurrency = 'GT';
-            feeCost = this.parseNumber (gtFee);
+        let fee = undefined;
+        if (gtFee !== undefined) {
+            if (gtFee === '0') {
+                feeCurrency = this.safeString (trade, 'fee_currency');
+                feeCost = this.safeString (trade, 'fee');
+            } else {
```
Comment (kroitor) [inline js/gateio.js:1864]: I think the proper way of handling this situation – is to add them both to `fees`, cause the fee may be paid partially in GT. We've seen this with the other exchanges before. Lmk if you can fix for that, if not I'll fix and merge.

Comment (frosty00) [reply js/gateio.js:1864]: yeah will fix later tonight

Comment (kroitor) [reply js/gateio.js:1864]: @frosty00  What the plan on this PR ?

Comment (frosty00) [reply js/gateio.js:1864]: will revise after midterm

**Pull request closed without merging**